modules/gapi/include/opencv2/gapi/garg.hpp

```diff
@@ -133,6 +133,48 @@ using GRunArgP = util::variant<
     >;
 using GRunArgsP = std::vector<GRunArgP>;
 
+//GAPI_EXPORTS cv::GRunArgsP bind(cv::GRunArgs &results);
+inline cv::GRunArgsP bind(cv::GRunArgs &results)
+{
+    cv::GRunArgsP outputs;
+    outputs.resize(results.size());
+    unsigned int i = 0;
+    for (auto && res_obj : results)
+    {
+        //auto &res_obj = std::get<0>(it);
+        // FIXME: this conversion should be unified
+        using T = cv::GRunArg;
+        switch (res_obj.index())
+        {
+#if !defined(GAPI_STANDALONE)
+        case T::index_of<cv::UMat>() :
+        {
+            outputs[i] = &(cv::util::get<cv::UMat>(res_obj));
+        }
+                                     break;
+#endif
+        case T::index_of<cv::Mat>() :
+        {
+            outputs[i] = &(cv::util::get<cv::Mat>(res_obj));
+        } break;
+        case T::index_of<cv::Scalar>() :
+            outputs[i] = &(cv::util::get<cv::Scalar>(res_obj));
+            break;
+        case T::index_of<cv::detail::VectorRef>() :
+            outputs[i] = cv::util::get<cv::detail::VectorRef>(res_obj);
+            break;
+        case T::index_of<cv::detail::OpaqueRef>() :
+            outputs[i] = cv::util::get<cv::detail::OpaqueRef>(res_obj);
+            break;
+        default:
+            GAPI_Assert(false && "This value type is not supported!"); // ...maybe because of STANDALONE mode.
+            break;
+        }
+        i++;
```

Comment:
instead of `i`, you should be able to use `outputs.emplace_back()`. The vector should be `reserve()`-d but not `resize()`-d then.

Reply:
done